frontend/src/_helpers/appUtils.js

```diff
@@ -398,26 +398,26 @@ export const executeAction = (_ref, event, mode, customVariables) => {
       case 'set-custom-variable': {
         const key = resolveReferences(event.key, _ref.state.currentState, undefined, customVariables);
         const value = resolveReferences(event.value, _ref.state.currentState, undefined, customVariables);
-        const customAppVariables = { ..._ref.state.currentState.variables };
-        customAppVariables[key] = value;
+        const customAppVariables = { ..._ref.state.currentState };
+        customAppVariables.variables[key] = value; // Updated object directly to get the variable value immediately
 
         return _ref.setState({
           currentState: {
             ..._ref.state.currentState,
-            variables: customAppVariables,
+            variables: customAppVariables.variables,
```

Comment:
Instead of mutating a key directly, can we do something like:

```
currentState: {
            ..._ref.state.currentState,
            variables: { ...currentState.variables, [key]: value },
}
```